### ISSUE TITLE
Accept regexp in group layer definitions

### ISSUE THREAD

**Issue by smontanari** [adamtornhill/code-maat#39]
When using code-maat on recent javascript (React) code bases unfortunately it's not possible to distinguish production code from test code, as the convention for React seems to be putting test files next to production files in the same folder. If I wanted to separate the production code from the test code in my code-maat analysis using the --group options I would liked to be able to write a file using regexp expressions to split the layer assignment, e.g.

```
app/frontend/.*/[^--]+\.js => Code
app/frontend/.*--test\.js => Unit tests
```

**Comment by adamtornhill** [comment]
Yes, the current grouping is too limited. In [CodeScene](https://codescene.io/) we use glob patterns instead, which works really well for this task.

The change you suggest should be fairly trivial to implement since code-maat already uses regular expressions for the group matching. However, code-maat also assumes the given `--group` expression is a path (that's the part we need to change).

I'd like to keep code-maat backwards compatible since I use the tool a lot in my book. I'd suggest we implement the feature above using a new command line flag.

**Comment by adamtornhill** [comment]
Solved for version 1.0.
